### ISSUE TITLE
CI failed on dashboards-search-relevance with TCP ECONNRESET

### ISSUE THREAD

**Issue by noCharger** [opensearch-project/opensearch-dashboards-functional-test#902]
### What is the bug?
_A clear and concise description of the bug._

CI failed on dashboards-search-relevance with TCP ECONNRESET

### How can one reproduce the bug?
_Steps to reproduce the behavior._

This only happening on CI. Ex: https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/6280962123/job/17359747615?pr=267

```
Received 25036087 of 25036087 (100.0%), 57.7 MBs/sec
Cache Size: ~24 MB (25036087 B)
/usr/bin/tar -z -xf /home/runner/work/_temp/30538c1d-45c3-42d2-a001-0847c6b2bfe6/cache.tgz -P -C /home/runner/work/dashboards-search-relevance/dashboards-search-relevance
Cache restored successfully
Received 4[19](https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/6280962123/job/17359747615?pr=267#step:21:20)43040 of 368128980 (11.4%), 40.0 MBs/sec
Received 96468992 of 368128980 (26.2%), 46.0 MBs/sec
Received 197132288 of 368128980 (53.5%), 62.2 MBs/sec
Received 251658240 of 368128980 (68.4%), 59.6 MBs/sec
Received 339738624 of 368128980 (92.3%), 64.5 MBs/sec
Received 368128980 of 368128980 (100.0%), 60.7 MBs/sec
Cache Size: ~351 MB (368128980 B)
/usr/bin/tar -z -xf /home/runner/work/_temp/aeea8250-ab99-4718-a8c5-9[20](https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/6280962123/job/17359747615?pr=267#step:21:21)4d2ea5034/cache.tgz -P -C /home/runner/work/dashboards-search-relevance/dashboards-search-relevance
Cache restored successfully
/opt/hostedtoolcache/node/14.20.1/x64/bin/npm ci

> cypress@9.5.4 postinstall /home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test/node_modules/cypress
> node index.js --exec install

Note: Overriding Cypress cache directory to: ~/.cache/Cypress

      Previous installs of Cypress may not be found.


Cypress 9.5.4 is installed in /home/runner/.cache/Cypress/9.5.4


> opensearch-dashboards-functional-test@3.0.0 postinstall /home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test
> husky install

husky - Git hooks installed
added 329 packages in 10.792s
/opt/hostedtoolcache/node/14.20.1/x64/bin/npx cypress cache list
┌─────────┬───────────────────┐
│ version │ last used         │
├─────────┼───────────────────┤
│ 13.2.0  │ a few seconds ago │
├─────────┼───────────────────┤
│ 13.3.0  │ 4 days ago        │
├─────────┼───────────────────┤
│ 9.5.4   │ a few seconds ago │
└─────────┴───────────────────┘
Using custom test command: yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js
run tests command "yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js"
current working directory "/home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test"
/opt/hostedtoolcache/node/14.20.1/x64/bin/yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js
yarn run v1.22.10
$ /home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test/node_modules/.bin/cypress run --env SECURITY_ENABLED=false --browser chrome --spec 'cypress/integration/plugins/search-relevance-dashboards/*.js'
[5560:1003/163830.175363:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is swiftshader, ANGLE is 
Couldn't determine Mocha version

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        9.5.4                                                                          │
  │ Browser:        Chrome 117 (headless)                                                          │
  │ Node Version:   v14.20.1 (/opt/hostedtoolcache/node/14.20.1/x64/bin/node)                      │
  │ Specs:          1 found (plugins/search-relevance-dashboards/1_query_compare.spec.js)          │
  │ Searched:       cypress/integration/plugins/search-relevance-dashboards/*.js                   │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  plugins/search-relevance-dashboards/1_query_compare.spec.js                     (1 of 1)
Error: read ECONNRESET
    at TCP.onStreamRead (node:internal/stream_base_commons:[21](https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/6280962123/job/17359747615?pr=267#step:21:22)1:20)
 {
  errno: -104,
  code: 'ECONNRESET',
  syscall: 'read'
}
Error: read ECONNRESET
    at TCP.onStreamRead (node:internal/stream_base_commons:211:20)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: The process '/opt/hostedtoolcache/node/14.20.1/x[64](https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/6280962123/job/17359747615?pr=267#step:21:65)/bin/yarn' failed with exit code 1
```

### What is the expected behavior?
_A clear and concise description of what you expected to happen._

CI pass

```
> opensearch-dashboards-functional-test@3.0.0 postinstall /home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test
> husky install

husky - Git hooks installed
added 335 packages in 10.61s
/opt/hostedtoolcache/node/14.20.1/x64/bin/npx cypress cache list
┌─────────┬───────────────────┐
│ version │ last used         │
├─────────┼───────────────────┤
│ 13.0.0  │ a few seconds ago │
├─────────┼───────────────────┤
│ 13.1.0  │ 6 days ago        │
├─────────┼───────────────────┤
│ 9.5.4   │ a few seconds ago │
└─────────┴───────────────────┘
Using custom test command: yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js
run tests command "yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js"
current working directory "/home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test"
/opt/hostedtoolcache/node/14.20.1/x64/bin/yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js
yarn run v1.22.10
$ /home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test/node_modules/.bin/cypress run --env SECURITY_ENABLED=false --browser chrome --spec 'cypress/integration/plugins/search-relevance-dashboards/*.js'
[4722:0906/232051.235940:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is swiftshader, ANGLE is 
Couldn't determine Mocha version

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        9.5.4                                                                          │
  │ Browser:        Chrome 116 (headless)                                                          │
  │ Node Version:   v14.20.1 (/opt/hostedtoolcache/node/14.20.1/x64/bin/node)                      │
  │ Specs:          1 found (plugins/search-relevance-dashboards/1_query_compare.spec.js)          │
  │ Searched:       cypress/integration/plugins/search-relevance-dashboards/*.js                   │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  plugins/search-relevance-dashboards/1_query_compare.spec.js                     (1 of 1)
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
Couldn't determine Mocha version


  Compare queries
    ✓ Should get comparison results (46233ms)


  1 passing (51s)
```

### What is your host/environment?
_Operating system, version._

ubuntu-latest, 11

### Do you have any additional context?
_Add any other context about the problem._

A potential change to ignore such failures

https://github.com/hilleer/vscode-yaml-plus-json/pull/146/files#diff-2785793ab42443fdaafae000e605c53af6013c0ae7bc2aa3a490ba23f9238d69

**Comment by SuZhou-Joe** [comment]
Hi @noCharger , it seems that the issue has something to do with build process of OSD core, and I created a dummy PR https://github.com/opensearch-project/dashboards-search-relevance/pull/316 in your repo and it turned out to work well, please retry to see if it has solved your problem.

**Comment by noCharger** [comment]
> the issue has something to do with build process of OSD core

@SuZhou-Joe Could you share more details about this build process change?

> I created a dummy PR https://github.com/opensearch-project/dashboards-search-relevance/pull/316 in your repo

I saw the CI passed, would you mind turn it into review so we can get it merged?

**Comment by SuZhou-Joe** [comment]
Actually what changes introduces in this PR have nothing to do with CI failure, it is due to infra changes in OSD core.

**Comment by ruanyl** [comment]
@SuZhou-Joe I suppose this workflow failure related to this issue as well? https://github.com/opensearch-project/opensearch-dashboards-functional-test/actions/runs/6466652988/job/17557912064?pr=736

**Comment by SuZhou-Joe** [comment]
> @SuZhou-Joe I suppose this workflow failure related to this issue as well? https://github.com/opensearch-project/opensearch-dashboards-functional-test/actions/runs/6466652988/job/17557912064?pr=736

I thought they were the same but actually not, search relevance CI runs on Node v14, which is OK when OSD start on `localhost`.

**Comment by SuZhou-Joe** [comment]
Should be fixed by #960 so I am going to close this issue.
